config-linux.md

```diff
@@ -455,6 +455,34 @@ The following parameters can be specified to set up the controller:
    }
 ```
 
+### <a name="configLinuxRDMA" />RDMA
+
+**`rdma`** (object, OPTIONAL) represents the cgroup subsystem `rdma`.
+For more information, see the kernel cgroups documentation about [rdma][cgroup-v1-rdma].
+
+The following parameters can be specified to set up the controller:
+
+* **`hca_device`** *(string, REQUIRED)* - specifies the device name whose resources limit to be configured
+* **`hca_handles`** *(uint32, REQUIRED)* - specifies the maximum number of hca_objects in the cgroup for a specified device
+* **`hca_objects`** *(uint32, REQUIRED)* - specifies the maximum number of hca_handles in the cgroup for a specified device
```

Comment:
You have “Default is "no limit".” in the Go comments, but REQUIRED here.  Reading the kernel docs, I think `hca_handles` and `hca_objects` should both be OPTIONAL.  If you like, you can add a requirement like [this][1]

> You MUST specify at least one of `hca_handles` or `hca_objects` in a given entry, and MAY specify both.

The kernel docs also make a distinction between leaving those unset (no change from the previous limit) or setting them to `max` (clearing a possible previous limit).  Since the spec supports joining and tweaking existing cgroups, I think we need a way to distinguish between the two cases.  The current policy seems to be using signed integers and [using -1 for “explicitly remove any limits”][2].

[1]: https://github.com/opencontainers/runtime-spec/blame/v1.0.1/config-linux.md#L338
[2]: https://github.com/opencontainers/runtime-spec/blame/v1.0.1/config-linux.md#L262

Reply:
You are right. I initially had it optional but than made it required to simplify the -1. But I think its fine as it already exist. 
I will make both of them as optional.
So when user doesn't want to configure any one of the value, it should set to -1. Correct?
Because at kernel level they are just optional KV pair, but at cgroups spec level it is a structure. So no_limit should be initialized as -1. Right?

Reply:
> So when user doesn't want to configure any one of the value, it should set to -1. Correct?

No, leave it unset to not change the limit (nothing written to the control file for this parameter).  Set to -1 to *clear* the limit (`max` written to the cgroup control file for this parameter).

Reply:
so when cgroups.New() is invokes, there are 4 valid values for hca_objects/hca_handles.
1. 0 - no allocation possible
2. +ve - valid positive value
3. -1 to clear the limits
4. do not touch the limits 

If we make these two parameters optional, what should user set in specs.LinuxRdma.HcaObjects to convey - 'do not touch the limits'.
Setting 0 will disable it (which is valid value).

Reply:
> ... what should user set in specs.LinuxRdma.HcaObjects to convey - 'do not touch the limits'.

`nil`, which is why [we want pointers][1].

[1]: https://github.com/opencontainers/runtime-spec/pull/942#discussion_r155593626

Reply:
ok. Sounds good. I will revise the spec.